gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -235,4 +233,42 @@ public Table create(String table, TableDefinition definition, BigQuery.TableOpti
   public BigQuery bigquery() {
     return bigquery;
   }
+
+  public static Builder builder(BigQuery bigquery, DatasetId datasetId) {
```

Comment:
We can probably package scope or remove this because we don't expect users to ever have to build a functional object from scratch. They'll just be handed one if they use the BigQuery service object at any point.  I believe the parallel builder method for Resource Manager's Project is package scoped, and I didn't include it in the Storage PR that's out right now.